src/PowerShellEditorServices/Language/FindDotSourcedVisitor.cs

```diff
@@ -32,15 +39,49 @@ public FindDotSourcedVisitor()
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitCommand(CommandAst commandAst)
         {
-            if (commandAst.InvocationOperator.Equals(TokenKind.Dot) &&
-                commandAst.CommandElements[0] is StringConstantExpressionAst)
+            CommandElementAst commandElementAst = commandAst.CommandElements[0];
+            if (commandAst.InvocationOperator.Equals(TokenKind.Dot))
             {
-                // Strip any quote characters off of the string
-                string fileName = commandAst.CommandElements[0].Extent.Text.Trim('\'', '"');
-                DotSourcedFiles.Add(fileName);
+                string path;
+                switch (commandElementAst)
+                {
+                    case StringConstantExpressionAst stringConstantExpressionAst:
+                        path = stringConstantExpressionAst.Value;
+                        break;
+
+                    case ExpandableStringExpressionAst expandableStringExpressionAst:
+                        path = GetPathFromExpandableStringExpression(expandableStringExpressionAst);
+                        break;
+
+                    default:
+                        path = null;
+                        break;
+                }
+
+                DotSourcedFiles.Add(PathUtils.NormalizePathSeparators(path));
             }
 
             return base.VisitCommand(commandAst);
         }
+
+        private string GetPathFromExpandableStringExpression(ExpandableStringExpressionAst expandableStringExpressionAst)
+        {
+            var path = expandableStringExpressionAst.Value;
+            foreach (var nestedExpression in expandableStringExpressionAst.NestedExpressions)
+            {
+                // If the string contains the variable $PSScriptRoot, we replace it with the corresponding value.
+                if (nestedExpression is VariableExpressionAst variableExpressionAst
+                    && variableExpressionAst.VariablePath.UserPath.Equals("PSScriptRoot", StringComparison.CurrentCultureIgnoreCase))
```

Comment:
Wouldn't `OrdinalIgnoreCase` be sufficient here?

Reply:
I think that's preferred -- it's what PowerShell uses

Reply:
Ok I'll change it.
It makes me wonder how the different [turkish I-s](https://en.wikipedia.org/wiki/Dotted_and_dotless_I) are handled in PowerShell variable names given that it's case insensitive.